streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -349,6 +359,57 @@ public String queryableStoreName() {
         return toStream().selectKey(mapper);
     }
 
+    @Override
+    public KTable<K, V> suppress(final Suppressed<K> suppressed) {
+        final SuppressedImpl<K> suppressImpl = (SuppressedImpl<K>) suppressed;
+        final String name = builder.newProcessorName(SUPPRESS_NAME);
+
+        final ProcessorSupplier<K, Change<V>> suppressionSupplier =
+            () -> new KTableSuppressProcessor<>(buildSuppress(suppressImpl));
+
+        final ProcessorParameters<K, Change<V>> processorParameters = new ProcessorParameters<>(
+            suppressionSupplier,
+            name
+        );
+
+        final ProcessorGraphNode<K, Change<V>> node = new ProcessorGraphNode<>(name, processorParameters, false);
+
+        builder.addGraphNode(this.streamsGraphNode, node);
+
+        return new KTableImpl<K, S, V>(
+            builder,
+            name,
+            suppressionSupplier,
+            keySerde,
+            valSerde,
+            Collections.singleton(this.name),
+            null,
+            false,
+            node
+        );
+    }
+
+    @SuppressWarnings("unchecked") // see the comments on the casts
+    private SuppressedImpl<K> buildSuppress(final SuppressedImpl<K> suppress) {
+        if (suppress.isFinalResultsSuppression()) {
+            // This throws an exception if the computation is not windowed
+            final long grace = findAndVerifyWindowGrace(streamsGraphNode);
+
+            // so we know we can cast the key to Windowed
+            final SuppressedImpl<? extends Windowed> windowedSuppress = (SuppressedImpl<? extends Windowed>) suppress;
+
+            final SuppressedImpl<? extends Windowed> finalResultsSuppression = SuppressedImpl.buildFinalResultsSuppression(
+                Duration.ofMillis(grace),
```

Comment:
The user is not capable of specifying a suppress time for final results suppressions (see `Suppressed.untilWindowClose`). Only a buffer config. The grace period is the only way to "set" the suppression time.

I'll rename the parameter to `gracePeriod` This is correct, since the time definition for final results mode uses the window end as the starting point.

Reply:
But I don't like that this wasn't immediately obvious, so I'll make some clarifying changes.

Reply:
> The grace period is the only way to "set" the suppression time.

Why this limitation? Let's say I have
```
KTable table = ... // windowed with some grace period X
KTable table2 = table.filter(...); // no suppression specified because I want to update filter for each base table update and want to be able to update with max delay X (== grace from above)
table.suppress(...).toString().forEach(); // I also want to react to a single update per key (ie, single final), but I want to have lower latency and want "single final update" after X/10 of the time -- I want to specify smaller grace period here
```

Does this make sense?

Reply:
Your example use case is legitimate, and it is indeed something that we sacrifice here. 
Allow me to paraphrase:
```
KTable table = ....windowedBy(TimeWindows.of(10).withGrace(100)).count(Materialized.as("count"));
table.suppress(untilWindowCloses(overrideGrace(10))).to("output");
```
This was a point of discussion early on in the KIP. The downside of this API is that querying "count" and observing "output", we will see divergent results, since "count" will permit some records that the suppression drops (say, any record that arrives more than 10ms later than its window).

We felt that if the operator's job is to emit only "final result" of the window's aggregation, then that's exactly what it should do. Redefining the window parameters is out of scope.

However, to your second comment, I didn't follow. We don't _just_ offer `untilWindowCloses`. You could alternatively do:

```
KTable table = ....windowedBy(TimeWindows.of(10).withGrace(100)).count(Materialized.as("count"));
table.suppress(untilTimeElapses(10)).to("output");
```

This won't give you "final results", since it will still emit updates if they are needed.

Final thought: Like I said, I don't think it's unreasonable what you proposed, but I think it's better to start with something simple and safe. If people are asking for this API later on, we can always add it.

Reply:
Ack. Thanks for clarification. It's a complex discussion... Sorry for repeating part of the KIP discussion here.

Reply:
No worries. There are indeed a lot of nuances to keep track of.